shoulder/model/x86_64/access_mechanism/cpuid.py

```diff
@@ -0,0 +1,52 @@
+#
+# Shoulder
+# Copyright (C) 2018 Assured Information Security, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from shoulder.model.access_mechanism import AbstractAccessMechanism
+from dataclasses import dataclass
+
+@dataclass()
+class CPUID(AbstractAccessMechanism):
+    """ Access mechanism for reading information from the CPUID instruction """
+
+    leaf: bytes = 0
+    """ The CPUID leaf to be read (eax)  """
+
+    output: str = "eax"
+    """ The output register that CPUID reads information into  """
+    """ Options: eax, ebx, ecx, edx  """
+
+    name: str = "cpuid"
+    """ The name of this access mechanism  """
+
+    def is_read(self):
+        return True
+
+    def is_write(self):
+        return False
+
+    def is_valid(self):
+        if str(output).lower() not in ["eax", "ebx", "ecx", "edx"]:
+            return False
+
+        # TODO: Check leaf/subleaf within valid range
```

Comment:
I'm fine leaving this as `TODO` for now, but I just wanted to make sure this wasn't intended to be implemented now

Reply:
Whoops, I didn't even mean to leave that comment in there. I think that this CPUID access mechanism model isn't quite perfect yet, are you ok with me removing the TODO in a future commit dedicated to revising this access mechanism?

Reply:
Yeah, that is fine! I figured the CPUID was still WIP and this TODO had to do with that, so we can do that later. I just wanted to make sure we weren't missing anything.